python/ray/tune/ray_trial_executor.py

```diff
@@ -275,7 +279,18 @@ def get_next_available_trial(self):
         # the first available result, and we want to guarantee that slower
         # trials (i.e. trials that run remotely) also get fairly reported.
         # See https://github.com/ray-project/ray/issues/4211 for details.
+        start = time.time()
         [result_id], _ = ray.wait(shuffled_results)
+        wait_time = time.time() - start
+        if wait_time > NONTRIVIAL_WAIT_TIME_THRESHOLD_S:
+            self._last_nontrivial_wait = time.time()
+        if time.time() - self._last_nontrivial_wait > BOTTLENECK_WARN_PERIOD_S:
```

Comment:
hm, if all `wait_time > NONTRIVIAL_WAIT_TIME_THRESHOLD_S`, then this message would never show up -- is this intended?

Reply:
My intuition here is that we should only warn if tune is constantly backlogged. If there was a wait that was slow, then probably it is not so.